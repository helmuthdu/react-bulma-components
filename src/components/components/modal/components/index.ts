export * from './modal-card';
export * from './modal-content';
