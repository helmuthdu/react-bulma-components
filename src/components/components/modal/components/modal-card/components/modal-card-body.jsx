import cn from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import modifiers from '../../../../../../modifiers';
import { Element } from '../../../../../layout/element';

export const ModalCardBody = ({ children, className, ...props }) => (
  <Element {...props} className={cn('modal-card-body', className)}>
    {children}
  </Element>
);

ModalCardBody.propTypes = {
  ...modifiers.propTypes,
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object,
  renderAs: PropTypes.string
};

ModalCardBody.defaultProps = {
  ...modifiers.defaultProps,
  renderAs: 'section'
};
