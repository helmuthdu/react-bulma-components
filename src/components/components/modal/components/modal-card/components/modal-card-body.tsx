import cn from 'classnames';
import React from 'react';
import modifiers, { Modifiers } from '../../../../../../modifiers';
import { Element } from '../../../../../elements';

type ModalCardBodyProps = Partial<Modifiers>;

export const ModalCardBody: React.FunctionComponent<ModalCardBodyProps> = ({
  children,
  className,
  ...props
}: ModalCardBodyProps) => (
  <Element {...props} className={cn('modal-card-body', className)}>
    {children}
  </Element>
);

ModalCardBody.defaultProps = {
  ...modifiers.defaultProps,
  renderAs: 'section'
};
