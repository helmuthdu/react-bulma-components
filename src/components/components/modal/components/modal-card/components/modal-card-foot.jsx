import cn from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import modifiers from '../../../../../../modifiers';
import { Element } from '../../../../../layout/element';

export const ModalCardFoot = ({ children, className, ...props }) => (
  <Element {...props} className={cn('modal-card-foot', className)}>
    {children}
  </Element>
);

ModalCardFoot.propTypes = {
  ...modifiers.propTypes,
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};

ModalCardFoot.defaultProps = {
  ...modifiers.defaultProps,
  renderAs: 'footer'
};
