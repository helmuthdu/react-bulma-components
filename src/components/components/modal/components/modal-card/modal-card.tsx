import cn from 'classnames';
import React from 'react';
import modifiers, { Modifiers } from '../../../../../modifiers';
import { Element } from '../../../../elements';

export * from './components';

type ModalCardProps = Partial<Modifiers> & {
  onClose?: (...args: any[]) => any;
};

export const ModalCard: React.FunctionComponent<ModalCardProps> = ({
  className,
  onClose,
  children,
  ...props
}: ModalCardProps) => (
  <Element {...props} className={cn('modal-card', className)}>
    {children}
  </Element>
);

ModalCard.defaultProps = {
  ...modifiers.defaultProps
};
