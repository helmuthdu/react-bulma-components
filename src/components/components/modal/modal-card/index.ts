export * from './modal-card';
export * from './modal-card-body';
export * from './modal-card-footer';
export * from './modal-card-header';
export * from './modal-card-title';
