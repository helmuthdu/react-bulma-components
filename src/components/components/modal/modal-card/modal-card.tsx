import cn from 'classnames';
import React from 'react';
import modifiers, { ElementModifier } from '../../../../modifiers';
import { Element } from '../../../elements';

type ModalCardProps = Partial<Omit<React.ComponentProps<'div'>, 'unselectable'>> &
  ElementModifier & {
    onClose?: (...args: any[]) => any;
  };

export const ModalCard: React.FunctionComponent<ModalCardProps> = ({
  className,
  onClose,
  children,
  ...props
}: ModalCardProps) => (
  <Element {...props} className={cn('modal-card', className)}>
    {children}
  </Element>
);

ModalCard.defaultProps = {
  ...modifiers.defaultProps
};
