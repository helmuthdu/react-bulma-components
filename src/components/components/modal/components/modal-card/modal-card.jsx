import cn from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import modifiers from '../../../../../modifiers';
import { Element } from '../../../../elements';
import { ModalCardBody, ModalCardFoot, ModalCardHead, ModalCardTitle } from './components';

// eslint-disable-next-line no-unused-vars
export const ModalCard = ({ className, onClose, children, ...props }) => (
  <Element {...props} className={cn('modal-card', className)}>
    {children}
  </Element>
);

ModalCard.Body = ModalCardBody;
ModalCard.Foot = ModalCardFoot;
ModalCard.Head = ModalCardHead;
ModalCard.Title = ModalCardTitle;

ModalCard.propTypes = {
  ...modifiers.propTypes,
  children: PropTypes.node,
  className: PropTypes.string,
  onClose: PropTypes.func,
  style: PropTypes.object
};

ModalCard.defaultProps = {
  ...modifiers.defaultProps
};
