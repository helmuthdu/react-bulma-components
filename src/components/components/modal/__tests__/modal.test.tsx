import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import { Modal } from '..';
import { ModalContent } from '../components';
import { ModalCard } from '../components/modal-card';
import { ModalCardHead, ModalCardTitle } from '../components/modal-card/components';

describe('Modal component', () => {
  it('should render modal-card-head', () => {
    const onClose = jest.fn();
    const { container, unmount } = render(
      <ModalCard onClose={onClose}>
        <ModalCardHead onClose={onClose}>
          <ModalCardTitle>Modal Title</ModalCardTitle>
        </ModalCardHead>
      </ModalCard>
    );
    fireEvent.click((container as any).querySelector('.delete'));
    expect(onClose).toHaveBeenCalledTimes(1);
    unmount();
  });

  it('should open the modal', () => {
    const onClose = jest.fn();
    const getComponent = show => (
      <Modal show={show} onClose={onClose}>
        <ModalCard>
          <ModalCardHead>
            <ModalCardTitle>Modal Title</ModalCardTitle>
          </ModalCardHead>
        </ModalCard>
      </Modal>
    );
    const { rerender, unmount } = render(getComponent(false));
    expect(window.document.querySelector('div.Modalis-active')).toBeNull();
    rerender(getComponent(true));
    expect(window.document.querySelector('div.Modalis-active')).toMatchSnapshot();
    unmount();
  });

  it('should close the modal', () => {
    const onClose = jest.fn();
    const { getByTestId, asFragment, unmount } = render(
      <Modal show onClose={onClose}>
        <ModalCard>
          <ModalCardHead onClose={onClose}>
            <ModalCardTitle>Modal Title</ModalCardTitle>
          </ModalCardHead>
        </ModalCard>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByTestId('modal-card-close-button'));
    expect(onClose).toHaveBeenCalledTimes(1);
    unmount();
  });

  it('should call close the modal on ESC key press', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal show onClose={onClose}>
        <ModalCard>
          <ModalCardHead onClose={onClose}>
            <ModalCardTitle>Modal Title</ModalCardTitle>
          </ModalCardHead>
        </ModalCard>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    const event = new KeyboardEvent('keydown', { keyCode: 27 } as any);
    document.dispatchEvent(event);
    expect(onClose).toHaveBeenCalledTimes(1);
    unmount();
  });

  it('should not close the modal on ESC key press', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal closeOnEsc={false} show onClose={onClose}>
        <ModalCard>
          <ModalCardHead>
            <ModalCardTitle>Modal Title</ModalCardTitle>
          </ModalCardHead>
        </ModalCard>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    const event = new KeyboardEvent('keydown', { keyCode: 27 } as any);
    document.dispatchEvent(event);
    expect(onClose).not.toHaveBeenCalled();
    unmount();
  });

  it('should not close the modal on other than ESC key press', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal show onClose={onClose}>
        <ModalCard>
          <ModalCardHead>
            <ModalCardTitle>Modal Title</ModalCardTitle>
          </ModalCardHead>
        </ModalCard>
      </Modal>
    );

    expect(asFragment()).toMatchSnapshot();
    const event = new KeyboardEvent('keydown', { keyCode: 23 } as any);
    document.dispatchEvent(event);
    expect(onClose).not.toHaveBeenCalled();
    unmount();
  });

  it('should render any child type', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal show onClose={onClose}>
        <div>CHILDREN</div>
        <div>CHILDREN</div>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it('should close the modal if clicked on background', () => {
    const onClose = jest.fn();
    const { getByTestId, asFragment, unmount } = render(
      <Modal show onClose={onClose} closeOnBlur>
        <ModalContent>Content</ModalContent>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByTestId('modal-background'));
    expect(onClose).toHaveBeenCalledTimes(1);
    unmount();
  });

  it('should not close the modal if clicked on background', () => {
    const onClose = jest.fn();
    const { getByTestId, asFragment, unmount } = render(
      <Modal show onClose={onClose} closeOnBlur={false}>
        <ModalContent>Content</ModalContent>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByTestId('modal-background'));
    expect(onClose).not.toHaveBeenCalled();
    unmount();
  });

  it('should not show close button', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal showClose={false} show onClose={onClose}>
        <ModalContent>Content</ModalContent>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it('should render empty because no document on scope', () => {
    const onClose = jest.fn();
    const { asFragment, unmount } = render(
      <Modal show onClose={onClose}>
        <ModalContent>Content</ModalContent>
      </Modal>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
});
