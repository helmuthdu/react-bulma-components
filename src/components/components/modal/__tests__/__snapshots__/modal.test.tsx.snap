// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should call close the modal on ESC key press 1`] = `<DocumentFragment />`;

exports[`Modal component should close the modal 1`] = `<DocumentFragment />`;

exports[`Modal component should close the modal if clicked on background 1`] = `<DocumentFragment />`;

exports[`Modal component should not close the modal if clicked on background 1`] = `<DocumentFragment />`;

exports[`Modal component should not close the modal on ESC key press 1`] = `<DocumentFragment />`;

exports[`Modal component should not close the modal on other than ESC key press 1`] = `<DocumentFragment />`;

exports[`Modal component should not show close button 1`] = `<DocumentFragment />`;

exports[`Modal component should open the modal 1`] = `null`;

exports[`Modal component should render any child type 1`] = `<DocumentFragment />`;

exports[`Modal component should render empty because no document on scope 1`] = `<DocumentFragment />`;
